### PR TITLE
GFORMS-986 format is mandatory for form components of type text

### PR DESCRIPTION
**Pull request by girishkamat**

### CODE REVIEW

File: test/uk/gov/hmrc/gform/sharedmodel/FieldValueFormatValueSpec.scala [hunk 1]
```diff
@@ -27,13 +27,14 @@ class FieldValueFormatValueSpec extends Spec {
...
-           "value": "${amountA + amountB}"
+           "value": "${amountA + amountB}",
+           "format": "text"
           }
       """)
     fieldValue should beJsSuccess(
       FormComponent(
         FormComponentId("sum"),
-        Text(ShortText.default, Add(FormCtx(FormComponentId("amountA")), FormCtx(FormComponentId("amountB")))),
+        Text(BasicText, Add(FormCtx(FormComponentId("amountA")), FormCtx(FormComponentId("amountB")))),
```
Comment (josef-vlach) [inline test/uk/gov/hmrc/gform/sharedmodel/FieldValueFormatValueSpec.scala:37]: `BasicText` is weird beast, do we really want to keep it?

 It is never used with `Text` as it is only constructed for multilines. Maybe we should mandate `format` on multiline as well and get rid of `BasicText`?

Note: It looks like `BasicText` is the same as `format: "text(0, 100000)", we should get rid of it.

Comment (girishkamat) [reply test/uk/gov/hmrc/gform/sharedmodel/FieldValueFormatValueSpec.scala:37]: Alex is fine with making it mandatory for TextArea. Will get rid of BasicText then. So for testing, format=text(0, 100000) should be ok then?

Comment (girishkamat) [reply test/uk/gov/hmrc/gform/sharedmodel/FieldValueFormatValueSpec.scala:37]: Removed BasicText and also updated docs at https://confluence.tools.tax.service.gov.uk/display/DF/Gforms+Text+Formats

File: app/uk/gov/hmrc/gform/core/parsers/FormatParser.scala [hunk 2]
```diff
@@ -127,8 +127,6 @@ object FormatParser {
         TextFormat(ShortText.default)
       } | "shortText(" ~ positiveInteger ~ "," ~ positiveInteger ~ ")" ^^ { (_, _, min, _, max, _) =>
         TextFormat(ShortText(min, max))
-      } | "text" ^^ { (_, _) =>
```
Comment (josef-vlach) [inline app/uk/gov/hmrc/gform/core/parsers/FormatParser.scala:130]: This is removing little to much.
I think this parser should stay, it should only produce `TextFormat(TextWithRestrictions(0, 100000))`.
Or did you agree with Alex that `"format":"text"` shouldn't be used?
Also removing of `BasicText` needs to propagate to frontend.

**Review by josef-vlach — CHANGES_REQUESTED** [review]

File: app/uk/gov/hmrc/gform/core/parsers/FormatParser.scala (hunk 2 above)
Comment (girishkamat) [reply app/uk/gov/hmrc/gform/core/parsers/FormatParser.scala:130]: I haven't agreed anything with Alex. I thought we were supposed to get rid of text. I will map it to TextFormat(TextWithRestrictions(0, 100000)), as ppl are used to that term

Comment (girishkamat) [reply app/uk/gov/hmrc/gform/core/parsers/FormatParser.scala:130]: "text" has been restored and mapped to `TextFormat(TextWithRestrictions(0, 100000))`

Comment (girishkamat) [reply app/uk/gov/hmrc/gform/core/parsers/FormatParser.scala:130]: Will create fontend PR for removal of BasicText

**Review by josef-vlach — APPROVED** [review]

**Pull request merged**